Spring Security + Login

cmn56690 opened the pull request:
Added additional security to endpoints which requires a generation of a JWT token to be used as the 'Bearer' token for each endpoint. (generated by passing a valid username + password to /login endpoint)

`
{
    "username": "Sawyer",
    "password": "password"
}
`

Updated login-page.component and added services and interceptor to interact with a few endpoints to log authenticate users before logging them in and navigating to /home. 

Generated JWT stored and accessible in LocalStorage and ownerId is set as a cookie to be used on /home page to return petsByOwnerId.

Commit:
Added Spring Security for endpoints and login

Commit:
Merge conflicts addressed

Commit:
Merge branch 'main' of https://github.com/ucmo-cs/Carebears_SP23 into jwtGeneration

Commit:
Angular service files for login

Commit:
Addressed login for petvax-app

Commit:
Merge branch 'main' of https://github.com/ucmo-cs/Carebears_SP23 into jwtGeneration

Commit:
Added remaining files to add findOwnerIdByUsername to return ownerId to be then set as a cookie upon login
Updated handleSubmit in login-page.component to login using username and password in users table and then call /userDetails?username= to capture the ownerId and set that as a cookie
JWT token is stored and accessible in LocalStorage

cmn56690 commented:
Need to do some clean-up before merging this, but this is functional code, if you want to checkout the branch.

Commit:
Added LoginResponse and corresponding files to return token as a json file.
Trying to add a call to /pets using the ownerId cookie, but receiving Refused Cookie error. Have commented that out for now.

luis-grados commented on petvax-app/petvax-app/src/app/components/login-page/login-page.component.ts:
```diff
@@ -1,21 +1,94 @@
+        if (token !== null) {
+          this.userService.getOwnerId(this.username, token).subscribe(
+            (response: any) => {
+              const ownerId = response.ownerId;
```
I think here might be const ownerId = response.ownerID. If anything I noticed when using cookies is that the value of the object must be the same as found in the API requests.

luis-grados commented on petvax-app/petvax-app/src/app/components/login-page/login-page.component.ts:
```diff
@@ -1,21 +1,94 @@
+          this.userService.getOwnerId(this.username, token).subscribe(
+            (response: any) => {
+              const ownerId = response.ownerId;
+              this.cookieService.set('ownerId', ownerId);
```
Same would apply here, should be: this.cookieService.set('ownerID', ownerId);

Commit:
Added pets.service and owner.service to make request to associated endpoints.
Updated home-page components to display data based upon responses from pets and owners.

Commit:
Removed unnecessary repository references.

Commit:
Removed @PreAuthorize annotation

Commit:
Removed unused libraries
